client/scripts/controllers/server/search.ts

```diff
@@ -0,0 +1,297 @@
+    } catch (e) {
+        console.error(e);
+        return [];
+    }
+  }
+
+  public searchThreadTitles = async (
+    searchTerm: string,
+    params: SearchParams
+  ): Promise<OffchainThread[]> => {
+    const { resultSize, chainScope, communityScope } = params;
+    try {
+      const response = await $.get(`${app.serverUrl()}/searchDiscussions`, {
+        chain: chainScope,
+        community: communityScope,
+        search: searchTerm,
+        results_size: resultSize,
+        thread_title_only: true,
+      });
+      if (response.status !== 'Success') {
+        throw new Error(`Got unsuccessful status: ${response.status}`);
+      }
+      return response.result.map((rawThread) => {
+        return modelFromServer(rawThread);
+      });
+    } catch (e) {
+      console.error(e);
+      return [];
+    }
+  };
+
+  public searchMentionableAddresses = async (
+    searchTerm: string,
+    params: SearchParams,
+    order?: string[]
+  ) => {
+    const { resultSize, communityScope, chainScope } = params;
+    try {
+      const response = await $.get(`${app.serverUrl()}/bulkAddresses`, {
+        chain: chainScope,
+        community: communityScope,
+        limit: resultSize,
+        searchTerm,
+        order,
+      });
+      if (response.status !== 'Success') {
+        throw new Error(`Got unsuccessful status: ${response.status}`);
+      }
+      return response.result;
+    } catch (e) {
+      console.error(e);
+      return [];
+    }
+  };
+
+  private sortResults = (a, b) => {
+    // TODO: Token-sorting approach
+    // Some users are not verified; we give them a default date of 1900
+    const aCreatedAt = moment(
+      a.created_at || a.createdAt || a.verified || '1900-01-01T:00:00:00Z'
+    );
+    const bCreatedAt = moment(
+      b.created_at || b.createdAt || b.verified || '1900-01-01T:00:00:00Z'
+    );
+    return bCreatedAt.diff(aCreatedAt);
+  };
+
+  public getHistory() {
+    const rawHistory = JSON.parse(localStorage.getItem(SEARCH_HISTORY_KEY)) || []
+    const history = []
+    for(let i = 0; i < rawHistory.length; i++){
+      history.push(SearchQuery.fromEncodedString(rawHistory[i]))
+    }
+    return history
+  }
+
+  public addToHistory(query: SearchQuery) {
+    this.removeFromHistory(query) // to ignore duplicates
+    const rawHistory = JSON.parse(localStorage.getItem(SEARCH_HISTORY_KEY)) || []
+    if(rawHistory.length >= SEARCH_HISTORY_SIZE){
+      rawHistory.pop()
+    }
+    rawHistory.unshift(query.toEncodedString())
+    localStorage.setItem(SEARCH_HISTORY_KEY, JSON.stringify(rawHistory))
+  }
+
+  public removeFromHistory(query: SearchQuery) {
+    const rawHistory = JSON.parse(localStorage.getItem(SEARCH_HISTORY_KEY)) || []
+    const index = rawHistory.indexOf(query.toEncodedString())
+    if(index > -1){
+      rawHistory.splice(index, 1)
+    }
+    localStorage.setItem(SEARCH_HISTORY_KEY, JSON.stringify(rawHistory))
+  }
+
+  public isValidQuery(searchQuery: SearchQuery){
```

Comment:
If I search for dates and numbers, I'll get preview results in the search bar, but am told it's an invalid query/can't navigate to full results. I know some of the valid query checks are holdovers from 1.0/2.0, so we can also punt this to a future PR.

![image](https://user-images.githubusercontent.com/22281010/146816216-3c83ce1c-ab1e-41e3-b8bc-225d9866c9a1.png)

Reply:
Fixed! We were filtering for only alphabet characters but I don't see why we need to do that.

Reply:
This is dope